src/components/Input.react.js

```diff
@@ -12,6 +13,7 @@ import {omit, isEmpty} from 'ramda';
 export default class Input extends Component {
     constructor(props) {
         super(props);
+        assertPropTypes(Input.propTypes, props, 'component prop', 'Input');
```

Comment:
I wonder if, rather than putting this assertion in the constructor of each component, we could do it for every component we create, and only when devtools are enabled

https://github.com/plotly/dash-renderer/blob/b9254ede45f232590fe3e53b2bf0e86c529759ed/src/TreeContainer.js#L66-L75

Reply:
yep, this is still in my mind, but I found the attributes and props world is much more complicated than imagined.